cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/flow/BasicFlowletContext.java

```diff
@@ -125,7 +124,8 @@ public int getInstanceId() {
   }
 
   public LoggingContext getLoggingContext() {
-    return new FlowletLoggingContext(getNamespaceId(), getApplicationId(), getFlowId(), getFlowletId());
+    return new FlowletLoggingContext(getNamespaceId(), getApplicationId(), getFlowId(), getFlowletId(),
+                                     getRunId().getId(), String.valueOf(getInstanceId()));
```

Comment:
The key for storing logs is derived using `FlowletLoggingContext`, and is not the exact key. So we can always rearrange the fields for the key.